shardingsphere-infra/shardingsphere-infra-context/src/main/java/org/apache/shardingsphere/infra/context/metadata/MetaDataContextsBuilder.java

```diff
@@ -94,21 +95,21 @@ public StandardMetaDataContexts build() throws SQLException {
             Collection<RuleConfiguration> ruleConfigs = schemaRuleConfigs.get(each);
             DatabaseType databaseType = DatabaseTypeRecognizer.getDatabaseType(dataSourceMap.values());
             Collection<ShardingSphereRule> rules = ShardingSphereRulesBuilder.buildSchemaRules(each, ruleConfigs, databaseType, dataSourceMap);
-            Map<Map<String, TableMetaData>, Map<String, TableMetaData>> tableMetaDataMap = SchemaBuilder.build(new SchemaBuilderMaterials(databaseType, dataSourceMap, rules, props));
+            TableMetaDatas tableMetaDatas = SchemaBuilder.build(new SchemaBuilderMaterials(databaseType, dataSourceMap, rules, props));
```

Comment:
Mayb a Map<LogicTableMetaData, ActualTableMetaData> is a concise variable to build? So that we can remove `TableMetaDatas` and `Map<Map<String, TableMetaData>, Map<String, TableMetaData>>`.

Reply:
But the `Map` only allows a `NULL` key, which may cause some tables to be missing. What do you think? Or something else like `Pair`?

Reply:
It is supposed that one actual tableMetadata has a corresponding logic tableMetaData, isn't it?

Suppose `ShardingSphereSchema` has all the `decorated` tableMetaData, the `OptimizeContextFactory` will own all the non-`decoreated` tableMetaData, right?

Reply:
Yeah, then I think it's fine. I'll fix him.